pilot/pkg/controllers/ipallocate/ipallocate.go

```diff
@@ -304,31 +311,55 @@ func (c *IPAllocator) markUsedOrQueueConflict(maybeMappedAddrs []netip.Addr, own
 	}
 }
 
-func (c *IPAllocator) statusPatchForAddresses(se *networkingv1alpha3.ServiceEntry, forcedReassign bool) ([]byte, error) {
+type jsonPatch struct {
+	Operation string      `json:"op"`
+	Path      string      `json:"path"`
+	Value     interface{} `json:"value"`
+}
+
+func (c *IPAllocator) statusPatchForAddresses(se *networkingv1alpha3.ServiceEntry, forcedReassign bool) ([]byte, []byte, error) {
 	if se == nil {
-		return nil, nil
+		return nil, nil, nil
 	}
 
 	existingAddresses := autoallocate.GetV2AddressesFromServiceEntry(se)
 	if len(existingAddresses) > 0 && !forcedReassign {
 		// this is likely a noop, but just to be safe we should check and potentially resolve conflict
 		c.markUsedOrQueueConflict(existingAddresses, config.NamespacedName(se))
-		return nil, nil // nothing to patch
+		return nil, nil, nil // nothing to patch
 	}
 	assignedAddresses := []apiv1alpha3.ServiceEntryAddress{}
 	for _, a := range c.nextAddresses(config.NamespacedName(se)) {
 		assignedAddresses = append(assignedAddresses, apiv1alpha3.ServiceEntryAddress{Value: a.String()})
 	}
 
-	addressUpdate := []jsonpatch.Operation{
+	replaceAddresses, err := json.Marshal([]jsonPatch{
 		{
 			Operation: "replace",
 			Path:      "/status/addresses",
 			Value:     assignedAddresses,
 		},
-	}
+	})
+
+	atomicAddAndAddresses, err2 := json.Marshal([]jsonPatch{
+		{
+			Operation: "test",
+			Path:      "/status",
+			Value:     nil, // we want to test that status is nil before we try to modiy it with this
```

Comment:
hmm, seems like there should be a way to do this with a single patch but I don't know of one

Reply:
I'll think on if this could be done better.

Reply:
not really fixing the patch process directly but we could adjust the crd so that status is never empty and instead defaults to `{}` which I think would mean that we never have to worry about adding it. This would also mean that the controller is only compatible with recent versions of our crd though... probably not ideal.

It also means that any service entry sitting around in the cluster would have `status: {}` at the end when status isn't being written which folks may oppose as well.